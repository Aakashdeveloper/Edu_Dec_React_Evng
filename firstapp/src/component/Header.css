header{
    background:tomato
}

.logo{
    color:white;
    font-size:30px;
    text-align:center
}