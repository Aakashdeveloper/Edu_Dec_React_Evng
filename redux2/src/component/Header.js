import React from 'react';
import {Link} from 'react-router-dom'

const Header = () => {
    return(
        <header>
            <div>
              <h1>Test  React APP</h1>
            </div>
        </header>
    )
}

export default Header;