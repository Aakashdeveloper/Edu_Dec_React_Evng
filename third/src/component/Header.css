header{
    background:tomato;
}

.logo{
    color:white;
    text-align:center;
    font-size:30px;
}