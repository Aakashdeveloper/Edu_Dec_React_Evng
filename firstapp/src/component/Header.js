import React,{Component} from 'react';
import './Header.css'

class Header extends Component{
    constructor(props){
        super(props)

        this.state={
            title:"React App",
            userInput:"User text Here"
        }
    }

    inputChanges(event){
        console.log(event.target.value)
        this.setState({userInput:event.target.value?event.target.value:'User text Here'})
        this.props.userText(event.target.value)
    }

    render(){
        return(
            <header>
               <div className="logo"
               onClick={()=>{console.log('logo clicked')}}>
                   {this.state.title}
               </div>
               <center>
                <input  onChange={this.inputChanges.bind(this)}/>
                <h3>{this.state.userInput}</h3>
               </center>
               <hr/>
            </header>
        ) 
    }
}

export default Header;