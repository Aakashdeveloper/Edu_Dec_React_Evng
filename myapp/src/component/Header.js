import React, {Component} from 'react';
import './Header.css'

class Header extends Component {

    constructor(props){
        super(props)

        this.state={
            title:'My React App',
            keywords:'User Input Here'
        }
    }

    inputChange(event){
        console.log(event.target.value)
        this.setState({keywords:event.target.value})
        this.props.newsSearch(event.target.value)
    }

    render(){
        return (
            <header>
                <div className="logo"
                onClick={()=> console.log("Clicked")}>
                    {this.state.title}
                </div>
                <center>
                  <input type="text"
                  onChange={this.inputChange.bind(this)}/> 
                  <p>{this.state.keywords}</p>
                </center>
                <hr/>
            </header>
        )
    }
}

export default Header;