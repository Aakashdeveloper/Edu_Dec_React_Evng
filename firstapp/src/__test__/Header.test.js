import React from 'react';
import Header from '../component/Header';
import {create} from 'react-test-renderer';

describe('Header Snapshot', () => {
    test("Header",() => {
        let tree = create(<Header/>);
        expect(tree.toJSON()).toMatchSnapshot();
    })
})
