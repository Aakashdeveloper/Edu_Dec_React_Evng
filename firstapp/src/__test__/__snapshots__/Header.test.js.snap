// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot Header 1`] = `
<header>
  <div
    className="logo"
    onClick={[Function]}
  >
    React App
  </div>
  <center>
    <input
      onChange={[Function]}
    />
    <h3>
      User text Here
    </h3>
  </center>
  <hr />
</header>
`;
